File: content/problems/0028-find-the-index-of-the-first-occurrence-in-a-string/solutions/naive-matching/Solution.kt
class Solution {
    fun strStr(haystack: String, needle: String): Int {
        for (start in 0..haystack.length - needle.length) {
            var index = 0
            while (index < needle.length && haystack[start + index] == needle[index]) index++
            if (index == needle.length) return start
        }
        return -1
    }
}

File: content/problems/0028-find-the-index-of-the-first-occurrence-in-a-string/solutions/naive-matching/solution.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        for (int start = 0; start + (int)needle.size() <= (int)haystack.size(); ++start) {
            int index = 0;
            while (index < (int)needle.size() && haystack[start + index] == needle[index]) ++index;
            if (index == (int)needle.size()) return start;
        }
        return -1;
    }
};
